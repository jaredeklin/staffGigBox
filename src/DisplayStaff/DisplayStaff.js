import React from 'react';
import PropTypes from 'prop-types';
import { Api } from '../Api/Api';

const DisplayStaff = props => {
  const {
    event,
    staffRole,
    deleteFromSchedule,
    handleEditClick,
    admin
  } = props;
  const api = new Api();

  const mapStaff = staffRole => {
    const { staff } = event;

    return staff
      .filter(staffMember => staffMember.role === staffRole)
      .map(person => {
        const { schedule_id, name } = person;

        return (
          <li key={schedule_id}>
            {name}
            {admin && (
              <div className="edit-container">
                <button
                  className="delete"
                  onClick={() => deleteFromSchedule(schedule_id)}
                />
                <button
                  className="edit"
                  onClick={() => handleEditClick(person)}
                />
              </div>
            )}
          </li>
        );
      });
  };

  const role =
    staffRole === 'Bartender' || staffRole === 'Barback'
      ? staffRole + 's'
      : staffRole;

  return (
    <ul className={api.getClassName(staffRole)}>
      <h4>{role}</h4>
      {mapStaff(staffRole)}
    </ul>
  );
};

export default DisplayStaff;

DisplayStaff.propTypes = {
  event: PropTypes.object,
  staffRole: PropTypes.string,
  admin: PropTypes.bool,
  deleteFromSchedule: PropTypes.func,
  handleEditClick: PropTypes.func
};
