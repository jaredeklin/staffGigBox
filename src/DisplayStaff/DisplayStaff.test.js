import React from 'react';
import { shallow } from 'enzyme';
import DisplayStaff from './DisplayStaff';

describe('DisplayStaff', () => {
  let wrapper;
  const mockEvent = { staff: [{ role: 'Bar Manager', staff_events_id: 34 }] };
  const mockDeleteSchedule = jest.fn();
  const mockHandleEditClick = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <DisplayStaff
        event={mockEvent}
        staffRole="Bar Manager"
        deleteFromSchedule={mockDeleteSchedule}
        handleEditClick={mockHandleEditClick}
        admin={true}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when the staffRole is Bartender', () => {
    wrapper = shallow(<DisplayStaff event={mockEvent} staffRole="Bartender" />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('click events', () => {
    it('should call deleteFromSchedule with correct params when clicked', () => {
      const expected = mockEvent.staff[0].staff_events_id;

      wrapper.find('.delete').simulate('click');
      expect(mockDeleteSchedule).toHaveBeenCalledWith(expected);
    });

    it('should call handleEditClick with correct params when clicked', () => {
      const expected = mockEvent.staff[0];

      wrapper.find('.edit').simulate('click');
      expect(mockHandleEditClick).toHaveBeenCalledWith(expected);
    });
  });
});
