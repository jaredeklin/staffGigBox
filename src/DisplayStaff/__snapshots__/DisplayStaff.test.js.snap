// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayStaff should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayStaff
    admin={true}
    deleteFromSchedule={[Function]}
    event={
        Object {
            "staff": Array [
              Object {
                "role": "Bar Manager",
                "staff_events_id": 34,
              },
            ],
          }
    }
    handleEditClick={[Function]}
    staffRole="Bar Manager"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Bar Manager
</h4>,
        Array [
          <li>
            <div
                        className="edit-container"
            >
                        <button
                                    className="delete"
                                    onClick={[Function]}
                        />
                        <button
                                    className="edit"
                                    onClick={[Function]}
                        />
            </div>
</li>,
        ],
      ],
      "className": "bar-manager",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bar Manager",
        },
        "ref": null,
        "rendered": "Bar Manager",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": "34Bar Manager",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <div
              className="edit-container"
>
              <button
                            className="delete"
                            onClick={[Function]}
              />
              <button
                            className="edit"
                            onClick={[Function]}
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="delete"
                  onClick={[Function]}
/>,
                <button
                  className="edit"
                  onClick={[Function]}
/>,
              ],
              "className": "edit-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "edit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Bar Manager
</h4>,
          Array [
            <li>
              <div
                            className="edit-container"
              >
                            <button
                                          className="delete"
                                          onClick={[Function]}
                            />
                            <button
                                          className="edit"
                                          onClick={[Function]}
                            />
              </div>
</li>,
          ],
        ],
        "className": "bar-manager",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bar Manager",
          },
          "ref": null,
          "rendered": "Bar Manager",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": "34Bar Manager",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <div
                className="edit-container"
>
                <button
                                className="delete"
                                onClick={[Function]}
                />
                <button
                                className="edit"
                                onClick={[Function]}
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="delete"
                    onClick={[Function]}
/>,
                  <button
                    className="edit"
                    onClick={[Function]}
/>,
                ],
                "className": "edit-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "delete",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "edit",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DisplayStaff should match the snapshot when the staffRole is Bartender 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayStaff
    event={
        Object {
            "staff": Array [
              Object {
                "role": "Bar Manager",
                "staff_events_id": 34,
              },
            ],
          }
    }
    staffRole="Bartender"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Bartenders
</h4>,
        Array [],
      ],
      "className": "bartenders",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bartenders",
        },
        "ref": null,
        "rendered": "Bartenders",
        "type": "h4",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Bartenders
</h4>,
          Array [],
        ],
        "className": "bartenders",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bartenders",
          },
          "ref": null,
          "rendered": "Bartenders",
          "type": "h4",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
